### PR TITLE
feat: /api/rerank endpoint + concurrent multi-model serving + KV cache fix

### PR DESCRIPTION
**Pull request by jaylfc**
## Summary

This PR adds two features for multi-model serving on RK3588:

### 1. `/api/rerank` endpoint with logit-based cross-encoder scoring

Implements a proper reranking endpoint for cross-encoder models like Qwen3-Reranker. Instead of generating text and parsing "yes"/"no" responses, it uses `RKLLM_INFER_GET_LOGITS` mode to extract the raw logit probabilities for "yes" and "no" tokens, computing a softmax relevance score (0-1).

**API format** (Jina/Cohere-compatible):
```
POST /api/rerank
{
  "model": "qwen3-reranker-0.6b",
  "query": "search query",
  "documents": ["doc1 text", "doc2 text"],
  "top_n": 5
}
```

**Tested:** Qwen3-Reranker-0.6B on RK3588 NPU produces scores of 0.978 for relevant documents and 0.025 for irrelevant, with proper discrimination.

### 2. Per-model locking and `--preload` for concurrent multi-model serving

Replaces the global `verrou` lock with per-model locks so requests to different models run truly in parallel. Each model gets its own NPU memory domain via `base_domain_id`.

**`--preload` flag:** Pre-loads specified models at startup:
```
rkllama_server --preload qwen3-embedding-0.6b,qwen3-reranker-0.6b,qmd-query-expansion
```

**Tested:** Concurrent embed (6s) + rerank (20s) requests complete in parallel instead of serialised. All 3 models preloaded at startup using 6.3GB total.

## Use Case

Running multiple QMD search pipeline models (embedding + reranking + query expansion) on a single RK3588 SBC. Without these changes, all requests are serialised through one global lock, and there's no reranking endpoint. With these changes, the full search pipeline can serve concurrent requests from multiple agents.

## Files Changed

| File | Change |
|------|--------|
| `src/rkllama/api/callback.py` | Capture logits from RKLLM callback |
| `src/rkllama/api/worker.py` | Add WORKER_TASK_RERANK, clean up per-model locks on stop |
| `src/rkllama/api/server_utils.py` | RerankEndpointHandler with prompt formatting, logit extraction, softmax scoring |
| `src/rkllama/api/variables.py` | Thread-safe per-model lock management |
| `src/rkllama/server/server.py` | `/api/rerank` + `/v1/rerank` endpoints, per-model locking, `--preload` flag |

## Testing

Tested on Orange Pi 5 Plus (RK3588, 16GB RAM, Armbian, RKLLM runtime v1.2.3, RKNPU driver v0.9.8) with:
- qwen3-embedding-0.6b (892MB, converted with RKLLM toolkit v1.2.3)
- qwen3-reranker-0.6b (892MB, converted with RKLLM toolkit v1.2.3)
- qmd-query-expansion-1.7B (2.3GB, fine-tuned from Qwen3-1.7B, converted with RKLLM toolkit v1.2.3)

## Backward Compatibility

- Without `--preload`, behavior is identical to before (lazy loading on first request)
- Per-model locking is strictly more permissive than the global lock
- Legacy `/generate` endpoint still uses the global lock

---

### Additional: KV cache fix for performance degradation

Clears KV cache after each embedding and rerank operation to prevent NPU performance degradation. Without this, embedding response times degrade from ~0.3s to 26s+ over hours of continuous operation.

### Additional: RuntimeMaxSec recommendation

For systemd deployments, recommend `RuntimeMaxSec=7200` (2 hour auto-restart) as a safety net.

### CODE REVIEW

**Comment by jaylfc** [comment]
**Additional commits since PR creation:**

- `c98c0ad` — Truncate rerank documents to 14000 chars to prevent 60s+ inference times on NPU. QMD sends pre-chunked text (~900 tokens) so this matches the useful context window.
- `e08f7f7` — Clear KV cache after embed/rerank operations to prevent NPU performance degradation (embed times degraded from 0.3s to 26s+ over hours without this fix).

**Embedding pipeline now verified at 100% completion** — all documents including 928KB chat transcripts fully embedded via NPU with batch optimisation and per-chunk retry logic.
